### PR TITLE
Cat Tinder API Validations

### PR DESCRIPTION
**Pull request by lundrew**

### CODE REVIEW

**Comment by lundrew** [comment]
Request to verify if we did the stretch challenges correctly.

File: spec/requests/cats_spec.rb [hunk 1]
```diff
@@ -84,6 +84,95 @@
...
+          name: 'Toast',
+          age: 2,
+          enjoys: 'walk',
+          image: 'http://www.catpics.com'
+        }
+      }
+      patch "/cats/#{cat.id}", params: update_cat_params
+      updated_cat = Cat.find(cat.id)
+      expect(response).to have_http_status(422)
+    end
```
Comment (WalkerTexas-coder) [inline spec/requests/cats_spec.rb]: Close! 
So the update should try to create an error
```ruby
it 'can't update an existing cat without a name' do
...
update_cat_params = {
        cat: {
          age: 2,
          enjoys: 'walk',
          image: 'http://www.catpics.com'
        }
      }
patch "/cats/#{cat.id}", params: update_cat_params
updated_cat = Cat.find(cat.id)
expect(response).to have_http_status(422)
json = JSON.parse(response.body)
expect(json['name']).to include "can't be blank"
```

**Review by WalkerTexas-coder — COMMENTED** [review]
left a comment with an example about how to do the update

File: spec/models/cat_spec.rb [hunk 2]
```diff
@@ -1,5 +1,24 @@
 require 'rails_helper'
 
 RSpec.describe Cat, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  it "should validate name" do
+    cat = Cat.create
```
Comment (elysemontano) [inline spec/models/cat_spec.rb]: What cat are you creating?  There will need to be attributes provided each time you create a cat.

**Review by elysemontano — CHANGES_REQUESTED** [review]
Changes needed for your model spec.

File: spec/models/cat_spec.rb [hunk 3]
```diff
@@ -1,5 +1,24 @@
...
+  it "should validate age" do
+    cat = Cat.create name: "Duke", enjoys: 'walking around the park', image: 'https://images.unsplash.com/photo-1529778873920-4da4926a72c2?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1036&q=80'
+    expect(cat.errors[:age]).to_not be_empty
+  end
+  it "should validate enjoys" do
+    cat = Cat.create name: "Duke", age: 5, image: 'https://images.unsplash.com/photo-1529778873920-4da4926a72c2?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1036&q=80'
+    expect(cat.errors[:enjoys]).to_not be_empty
+  end
+  it "should validate enjoys is more than 10 characters" do
+    cat = Cat.create enjoys: "Walks"
```
Comment (elysemontano) [inline spec/models/cat_spec.rb]: Still need the rest of the parameters when creating this cat.

File: spec/requests/cats_spec.rb [hunk 4]
```diff
@@ -84,6 +84,97 @@
...
+          age: 2,
+          enjoys: 'walk',
+          image: 'http://www.catpics.com'
+        }
+      }
+      patch "/cats/#{cat.id}", params: update_cat_params
+      updated_cat = Cat.find(cat.id)
+      expect(response).to have_http_status(422)
+      json = JSON.parse(response.body)
+      expect(json['name']).to include "can't be blank"
```
Comment (elysemontano) [inline spec/requests/cats_spec.rb:172]: You should have a test for each attribute for update as well.

**Review by elysemontano — CHANGES_REQUESTED** [review]
A few more changes in your tests.  Take a look at the comments.

**Review by sjproctor — APPROVED** [review]

**Review by sjproctor — APPROVED** [review]

**Review by elysemontano — APPROVED** [review]

**Pull request merged**